### PR TITLE
test: fy-currency component inside add-edit-adv-request page

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=oKLDnzY)
### <samp>🤖 Generated by Copilot at 8f1f1cb</samp>

Enabled and updated the test suite for the `FyCurrencyComponent` component. Imported the necessary modules and mock data to test the component's functionality and interaction with the modal controller and the currency chooser component.

### <samp>🤖 Generated by Copilot at 8f1f1cb</samp>

> _`xdescribe` no more_
> _Many imports for testing_
> _Winter of mock data_

### <samp>🤖 Generated by Copilot at 8f1f1cb</samp>

*  Enable and update the test suite for the `FyCurrencyComponent` component ([link](https://github.com/fylein/fyle-mobile-app/pull/2517/files?diff=unified&w=0#diff-efcea4d32c319a8225ea422a365244c59b659009df628a2c7af98ee39ff25e18L1-R161))

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-advance-request/fy-currency/fy-currency.component.spec.ts [hunk 1]
```diff
@@ -1,26 +1,162 @@
...
+      //@ts-ignore
+      component.ngControl.valid = false;
+      expect(component.valid).toBeTrue();
+    });
+  });
+
+  it('get value(): should return innerValue', () => {
+    //@ts-ignore
+    component.innerValue = mockCurrency;
+    expect(component.value).toEqual(mockCurrency);
```
Comment (shishirams222) [inline src/app/fyle/add-edit-advance-request/fy-currency/fy-currency.component.spec.ts:93]: Shouldn't we be creating variable, set the result of get Value() method to that variable and then check?

File: src/app/fyle/add-edit-advance-request/fy-currency/fy-currency.component.spec.ts [hunk 2]
```diff
@@ -1,26 +1,162 @@
...
+    const onChangeCallbackSpy = spyOn(component, 'onChangeCallback');
+    component.value = mockCurrency;
+    expect(component.value).toEqual(mockCurrency);
+    expect(formValueSpy).toHaveBeenCalledOnceWith(mockCurrency);
+    expect(onChangeCallbackSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('convertInnerValueToFormValue(): should return amount as null and currency as homeCurrency if inner value is not defined', () => {
+    component.homeCurrency = 'INR';
+    expect(component.convertInnerValueToFormValue(undefined)).toEqual({
```
Comment (shishirams222) [inline src/app/fyle/add-edit-advance-request/fy-currency/fy-currency.component.spec.ts:108]: same as above. Shouldn't we call the method `convertInnerValueToFormValue()` with right set of params to check the result?

**Review by shishirams222 — APPROVED** [review]
Two comments. Rest of the cases lg. Please get green from Dhar too for the tests that I have added comments

File: src/app/fyle/add-edit-advance-request/fy-currency/fy-currency.component.spec.ts (hunk 1 above)
Comment (suyashpatil78) [reply src/app/fyle/add-edit-advance-request/fy-currency/fy-currency.component.spec.ts:93]: Yes, changing this...

**Review by Chethan-Fyle — APPROVED** [review]

**Review by OmkarJ13 — APPROVED** [review]

**Review by Julias0 — APPROVED** [review]

**Pull request merged**